pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -1484,6 +1569,11 @@ func buildListener(opts buildListenerOpts) *xdsapi.Listener {
 		listenerFilters = append(listenerFilters, listener.ListenerFilter{Name: envoyListenerTLSInspector})
 	}
 
+	if opts.needHTTPInspector {
```

Comment:
Need proxy version check.

Reply:
this was marked as resolved but doesn't appear to be. Can you explain why you didn't add the check?

Reply:
@crazyxy please address the concerns here.

Reply:
You can ignore this. This value will never be set for anything other than istio 1.3+, by the calling code way up in the stack. ListenerProtocolAuto is the only logic that sets this to true. and ListenerProtocol is Auto only when the modelProtocoltoListenerProtocol returns auto (which is guarded by proxy version check)